### ISSUE TITLE
Long names in folder list 

### ISSUE THREAD

**Issue by MarcusWolschon** [k9mail/k-9#5290]
The old folder list (in play store) show deep subfolder pathes in a sensible manner and allow searching for a specific folder / switching between the visibility classes.

With the GitHub version of K9 all folders have the same name of the parent folders with no front/mid elipsis , no horizontal scrolling, no collapsible tree. 

![Screenshot_20210509-110948](https://user-images.githubusercontent.com/604692/117566643-118f3100-b0b8-11eb-8f42-14ba273693b9.png)

**Comment by MarcusWolschon** [comment]
A very first mitigation (not solution) would be to make them elipsize=midde.
The code seems to be in K9Drawer.kt in

```
val drawerItem = PrimaryDrawerItem().apply {
                iconRes = folderIconProvider.getFolderIcon(folder.type)
                identifier = drawerId
                tag = folder
                nameText = getFolderDisplayName(folder)
                displayFolder.unreadCount.takeIf { it > 0 }?.let {
                    badgeText = it.toString()
                    badgeStyle = folderBadgeStyle
                }
                selectedColorInt = selectedBackgroundColor
                textColor = selectedTextColor
            }
            ```

However I can't yet find out how to set elipsize for  [mikepenz /
MaterialDrawer ](https://github.com/mikepenz/MaterialDrawer/issues/2311).
It seems to have been fixed to "end"  in Issue [#2311](https://github.com/mikepenz/MaterialDrawer/issues/2311)

**Comment by MarcusWolschon** [comment]
Created a first mitigation and a pull request to make it usable right away .
https://github.com/k9mail/k-9/pull/5291

**Comment by cketti** [comment]
Closed via ec8efc631e4a480b529d0545e2e2d15bcae18628
